### PR TITLE
Display a hint in the navigation menu if the tags list is empty

### LINKED ISSUE #913: Add hint text to navigation drawer
Currently, when no tags exist, the tags section is hidden in the navigation drawer. Let's add some hint text in place of it to indicate that tags will appear there.

Screenshot 2019-12-17 at 16 55 55

### PR DESCRIPTION
**Pull request by hichamboushaba**
### Fix
Fixes #913 by displaying a hint when the tags list is empty:

I had to use a separate layout and add the divider + textview separately because the NavigationView doesn't allow a lot of customization of the items, and it uses the same style for all entries.

Also, I'm not really sure about the style of the TextView, @SylvesterWilmott please let me know if I need to change it.

#### Display hint if there is no tags
1. Remove all tags.
2. Open the navigation drawer.
3. Confirm that the hint is visible.

#### Hide the hint when there is some tags
1. Add some tags.
2. Open the navigation drawer.
3. Confirm that the hint and the divider are hidden.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/src/main/res/values/dimens.xml
Comment (hichamboushaba) [inline Simplenote/src/main/res/values/dimens.xml:57]: This value is to reuse the same elevation from the material library: https://cs.android.com/android-studio/platform/tools/adt/idea/+/mirror-goog-studio-master-dev:designer/testData/property/material.xml;l=379?q=design_navigation_elevation&sq=

File: Simplenote/src/main/res/layout/activity_notes.xml
Comment (theck13) [inline Simplenote/src/main/res/layout/activity_notes.xml:53]:
```suggestion
            android:background="?attr/drawerBackgroundColor"
            app:elevation="0dp"
```

We'll want to keep the `android:background="?attr/drawerBackgroundColor"` attribute and value for the `NavigationView` view so that the background color matches the rest of the navigation drawer.  It's not noticeable with the **_Light_** theme since it uses the default background color, but it's apparent with the **_Dark_** theme.  It's more apparent with other styles too.  See the screenshots below for illustration.

1249_navigation_tags_hint_portrait

File: Simplenote/src/main/res/layout/activity_notes.xml
Comment (theck13) [inline Simplenote/src/main/res/layout/activity_notes.xml]:
```suggestion
            android:layout_marginEnd="@dimen/navigation_margin"
            android:layout_marginStart="@dimen/navigation_margin"
            android:paddingBottom="@dimen/padding_large"
            android:paddingTop="@dimen/padding_large"
```

Let's vertically align the hint text with the navigation drawer items above it.  It looks like the left/start and right/end is 22dp.  The top and bottom padding look fine to me with 16dp.  See the screenshots below for illustration.

1249_navigation_tags_hint_layout

File: Simplenote/src/main/res/values/dimens.xml
Comment (theck13) [inline Simplenote/src/main/res/values/dimens.xml:57]:
```suggestion
    <dimen name="navigation_elevation">16dp</dimen>
    <dimen name="navigation_margin">22dp</dimen>
```

The `navigation_margin` dimension could be used for the navigation drawer margin mentioned above.

File: Simplenote/src/main/res/layout/activity_notes.xml
Comment (theck13) [inline Simplenote/src/main/res/layout/activity_notes.xml:72]:
```suggestion
            android:text="@string/empty_tags_hint"
            android:textColor="?attr/notePreviewColor" />
```

The default text color looks fine.  If we want to have more control over that color, we could add the `android:textColor="?attr/notePreviewColor"` attribute and value to the `TextView` view.  I'll leave that decision to @SylvesterWilmott though.  See the screenshots below for illustration.

1249_navigation_tags_hint_color

**Review by theck13 — CHANGES_REQUESTED** [review]

File: Simplenote/src/main/res/layout/activity_notes.xml
Comment (hichamboushaba) [reply Simplenote/src/main/res/layout/activity_notes.xml:53]: Nice catch 👍

**Review by theck13 — COMMENTED** [review]
The changes look good.  I'm going to wait for @SylvesterWilmott's review before merging.

**Comment by hichamboushaba** [comment]
@SylvesterWilmott can you please review this PR? today is my last day in my rotation, and I hope I can finish it before the eod 🙂

**Comment by SylvesterWilmott** [comment]
Looks good to me! Sorry for the delay on this one.

**Review by SylvesterWilmott — APPROVED** [review]

**Review by theck13 — CHANGES_REQUESTED** [review]
Some conflicts need to be resolved.  Once they are, we can merge this.

**Review by theck13 — APPROVED** [review]

**Pull request merged**
